### DISCUSSION TITLE
MITMPROXY does not pick up all the HTTPS/HTTP traffics in transparent mode.

### DISCUSSION THREAD

**Discussion by nguyenhoangdai** [mitmproxy/mitmproxy#5499] [Intercept Everything!]
I use Transparent Mode on mitmproxy and configure all the traffics to redirect to port 8080 as instructions. Mitmproxy does not pick up all the traffics.
One example is that if I access the page that also loads ReCAPTCHA from google. Mitmproxy does not pick up the request to google for the recaptcha which I see it is captured in the DEV tools. I'm also aware of --anticache option but it doesn't help either.
I guess my question is how can I actually pick up all HTTPS/HTTP traffics in mitmproxy? I'd appreciate any recommendation or explanation you could give.

System information:
Mitmproxy: 8.1.1
Python: 3.9.13
Open SSL: OpenSSL 3.0.5 5 Jul 2022
Platform: windows-10-10.0.19044-SP0

_Originally posted by @nguyenhoangdai in https://github.com/mitmproxy/mitmproxy/issues/5453#issuecomment-1197042893_

**Comment by nguyenhoangdai** [comment]
Please see the provided screen-recording for re-producing. I also attach the traffic logs (zip file) from mitmproxy and dev-tools for this session. 
For quick demonstration, please see the screenshot which showing the POST request made to google for recaptcha which is not captured by mitmproxy.
https://user-images.githubusercontent.com/16288442/181342633-954460f5-80b6-41e5-a7bb-d1267f8369dc.mp4

[traffic.zip](https://github.com/mitmproxy/mitmproxy/files/9202486/traffic.zip)
![screenshot](https://user-images.githubusercontent.com/16288442/181343613-f025a147-47f7-433c-8293-746cc7cc62ab.PNG)

**Comment by mhils** [comment]
If you are just interested in your browser's traffic, I would strongly recommend to just configure a regular proxy and _not_ use transparent mode. If the requests do not show up in mitmproxy, it's very likely that you have an issue in your iptables configuration. You cannot get requests through mitmproxy without them showing up in the UI.

**Comment by ericbeland** [comment]
Could it be because it is HTTP/3 and UDP? 

I know google is ahead of the curve on deploying HTTP/3 as they are sponsoring it, can control the browser, and can drive their own costs down by forcing requests to be more efficient.

**Comment by nguyenhoangdai** [comment]
@mhils Thanks for your response. I'm not just interested in my browser's traffic. I just used it as a demonstration. Just to clarification, I used transparent mode to monitor the traffics from my test device. Here is what I did to set up the transparent mode:

1 - Enable IP forwarding.(I'm using windows10 so I set IPEnableRouter to 1 and reroot after setting the value)

2- Disable ICMP redirects ( using netsh command)

3- Redirect all traffics from port 80 & 443 to 8080 (also using netsh command)

4 - MITMPROXY certificate installed 

I assume if there was any misconfiguration. MITMPROXY wouldn't pick up any traffics at all. Here MITMPROXY is picking traffics but just missing some!

**Comment by nguyenhoangdai** [comment]
Hi @ericbeland Wow yeah that could be it. I just double checked on the traffics and surprisingly the missed traffics are the one with H3 protocol. Please see in the screenshot!
<img width="1497" alt="Screen Shot 2022-07-28 at 12 04 54 PM" src="https://user-images.githubusercontent.com/16288442/181596381-6081e7fb-61b6-4156-8ff6-dd0d57d29dc4.png">

**Comment by nguyenhoangdai** [comment]
Update: I disable QUIC protocol on Chrome and all the traffics are forced to use H2 or below protocol. MITMPROXY is able to pick up all the traffics. Thanks a lot for your suggestion. 
https://support.eset.com/en/kb6757-disable-quic-protocol-in-google-chrome-browser
